### PR TITLE
Auto GROUP BY in the DataExplorer

### PR DESCRIPTION
**Pull request by 121watts**
- [x] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergable
  - [x] Tests pass
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

Connect #2390 

### The problem
We want the power of auto group by `:interval:` in the DataExplorer

### The Solution
Add it!

### CODE REVIEW

File: ui/src/shared/components/DatabaseList.js [hunk 1]
```diff
@@ -49,9 +49,12 @@ const DatabaseList = React.createClass({
...
     const differentSource = !_.isEqual(prevSource, nextSource)
 
+    if (prevQuery.rawText === nextQuery.rawText) {
+      return
+    }
+
     const newMetaQuery =
-      prevQuery.rawText !== nextQuery.rawText &&
-      nextQuery.rawText.match(/^(create|drop)/i)
+      nextQuery.rawText && nextQuery.rawText.match(/^(create|drop)/i)
```
Comment (cryptoquick) [inline ui/src/shared/components/DatabaseList.js]: Nice RegEx, LGTM

File: ui/src/data_explorer/containers/DataExplorer.js [hunk 2]
```diff
@@ -122,12 +121,13 @@ class DataExplorer extends Component {
...
             timeRange={timeRange}
             activeQuery={this.getActiveQuery()}
-            initialGroupByTime={INITIAL_GROUP_BY_TIME}
+            initialGroupByTime={AUTO_GROUP_BY}
           />
           <Visualization
             views={VIS_VIEWS}
             activeQueryIndex={0}
             timeRange={timeRange}
+            templates={TEMPLATES}
```
Comment (cryptoquick) [inline ui/src/data_explorer/containers/DataExplorer.js]: This doesn't create a new one each time, so if this is changed elsewhere in the app, the change might persist / conflict with state, and further, it doesn't scale to multiple items.

Comment (cryptoquick) [reply ui/src/data_explorer/containers/DataExplorer.js]: Come to think of it, unless you have a good reason we shouldn't, to be safe, let's `[...TEMPLATES]` here.

Comment (121watts) [reply ui/src/data_explorer/containers/DataExplorer.js]: That's the beauty of the way the `:interval:` template works, its static.

**Pull request merged**